src/Symfony/Component/Console/Command/Command.php

```diff
@@ -250,12 +253,22 @@ public function run(InputInterface $input, OutputInterface $output)
 
         $input->validate();
 
-        if ($this->code) {
-            $statusCode = call_user_func($this->code, $input, $output);
-        } else {
-            $statusCode = $this->execute($input, $output);
+        try {
+            if (!$this->lock($output)) {
+                $statusCode = 0;
+            } elseif ($this->code) {
+                $statusCode = call_user_func($this->code, $input, $output);
+            } else {
+                $statusCode = $this->execute($input, $output);
+            }
+        } catch (\Exception $e) {
```

Comment:
That won't work for php 5.5 and 5.6, what you can do is catch the `\Exception` first and then the throwable or error.